toolkit/tools/pkg/imagecustomizerlib/liveosisobuilder.go

```diff
@@ -542,7 +605,15 @@ func (b *LiveOSIsoBuilder) prepareLiveOSDir(writeableRootfsDir string, isoMakerA
 		return err
 	}
 
-	err = b.updateGrubCfg(b.artifacts.grubCfgPath, extraCommandLine)
+	if inputIsoSavedKernelArgsFilePath != "" {
```

Comment:
This code seems weird. Shouldn't the path where the file is saved always be the same, regardless of if there is an existing file or not?

Reply:
The existing file we checked was the file from the input iso. If such a file exists, we use it and update it from where it is. Otherwise, we'll create a new file at the to-be-created-iso artifacts folder.
I've updated the code such that we check if the input iso file is present, and if so, we copy it to the to-be-created-iso artifacts folders. And then update that new copy. This is cleaner as we no longer modify the contents of the expanded input iso.

Reply:
fixed.